Connect quizzes

km322 opened the pull request:
## Tracking Info

Resolves #68 

*NOTE: this does not account for module 4*

## Changes

<!-- What changes did you make? -->

- fixed the format of the quizzes and moved them so the quizzes are connected
- added module gates for the quizzes. 

## Testing

<!-- How did you confirm your changes worked? -->

- logged into my account on Safari and tried to access the quizzes and attempted the quizzes.

## Confirmation of Change

<!-- Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change. -->

https://github.com/user-attachments/assets/9abff190-3b47-4ba3-bd12-0ee8a6f8a9c2

eshaan-s18 reviewed (approved):
Fantastic job on this! This is so exciting to see everything coming together! I only noticed this small issue:

1. Module 1 quiz isn't navigating to the right page when I click on the start quiz button:
![Screenshot 2025-06-27 at 4 33 47 PM](https://github.com/user-attachments/assets/60f43b55-bc8c-4011-a0ec-81629733a954)

Other than that this is awesome. Great work! I'll approve since its just a small change to make

km322 commented:
fixed!

OrigamiStarz reviewed (dismissed):
Hey Ketan, everything looks great! One thing we could consider adding to the quizzes is to have a warning before users can refresh the page inside of the quizzes (similar to the pop up that is triggered when the exit button is pressed, except it is for when the user is refreshing in the quiz page). Otherwise, I found that the quizzes work perfectly as intended. 

Some other unrelated issues I found while testing:
1. When I wanted to test the backend server being down, I found that I couldn't sign in (which is expected), but also that clicking the sign in button did not notify the user that the server was down. The user would probably be very confused since the sign in button did nothing, so I think adding an error state would be ideal. 
2. Some minor style issues related to Chrome settings font size not set to medium (for reference, mine is set to large):
![image](https://github.com/user-attachments/assets/d0781588-3df1-4512-9b56-22e151ccc89f)
![image](https://github.com/user-attachments/assets/084df6ca-12ee-416b-8ae7-6e3a7f41bef9)

Overall, I couldn't really find any issues with your implementation for the quizzes, so I'm approving this. Fantastic work!

km322 commented:
Ok!

I added that "are u sure reload" u see when u reload mid quiz, I also fixed the module 3 clipping. 

The other issues will probably need another GitHub issue or something
